WordSearch.java

```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class WordSearch {
+    private List<String> results = new ArrayList<>();
+
+    public void findAllWords(int col, int row, char[][] characters, Dictionary dictionary) {
+        for (int i = 0; i < characters.length; i++) {
+            for (int j = 0; j < characters[1].length; j++) {
+                String word = String.valueOf(characters[i][j]);
+                if (dictionary.isPrefix(word)) {
+                    boolean[][] visit = new boolean[characters.length][characters[1].length];

```

Comment:
You should also replace `characters.length` and `characters[1].length` with `row` and `col`.